benches/engines/mod.rs

```diff
@@ -0,0 +1,238 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+extern crate criterion;
+extern crate kvproto;
+extern crate test_util;
+extern crate tikv;
+
+use criterion::{black_box, Bencher, Criterion};
+use kvproto::kvrpcpb::Context;
+use std::fmt;
+use test_util::generate_random_kvs;
+use tikv::storage::engine::{
+    BTreeEngine, Engine, Modify, RocksEngine, Snapshot, TestEngineBuilder,
+};
+use tikv::storage::{Key, CF_DEFAULT};
+
+const DEFAULT_KEY_LENGTH: usize = 64;
+const DEFAULT_ITERATIONS: usize = 1000;
+
+trait EngineFactory<E: Engine>: Clone + fmt::Debug + 'static {
+    fn build(&self) -> E;
+}
+
+#[derive(Clone)]
```

Comment:
You can derive Copy as well, so that no need to write `engine.clone()` (instead, just `engine`)